lib/view_component/base.rb

```diff
@@ -37,7 +38,7 @@ class Base < ActionView::Base
     # For example, one might compile component-specific assets at this point.
     # @private TODO: add documentation
     def self._after_compile
-      # noop
+      super if defined?(super)
```

Comment:
Do we need this change?

Reply:
Hey @elia, this is _super_ (haha, get it?!) interesting! First, I assume by `AC::Base` you actually mean `VC::Base`, is that right?

I think what you've got here is fine, although it makes me a little nervous that some future module may forget to call `super` and therefore interfere with `Translatable`'s functionality. Fortunately our tests should catch something like that.

It looks like we're uncovering the fact that relying on the inheritance hierarchy to manage lifecycle events is inherently brittle. It would probably be easier and more flexible to implement them using something like ActiveModel's callbacks API. Something to think about for the future.

Reply:
> It looks like we're uncovering the fact that relying on the inheritance hierarchy to manage lifecycle events is inherently brittle. It would probably be easier and more flexible to implement them using something like ActiveModel's callbacks API. Something to think about for the future.

I think that's a good idea. What are y'alls thoughts on adding a hook to the compiler itself so that we can use it for translations and keep translation compilation logic out of components themselves?

Reply:
I like it!

Reply:
> Hey @elia, this is super (haha, get it?!) interesting! First, I assume by AC::Base you actually mean VC::Base, is that right?

Indeed! Corrected the comment, thanks!

> It looks like we're uncovering the fact that relying on the inheritance hierarchy to manage lifecycle events is inherently brittle. It would probably be easier and more flexible to implement them using something like ActiveModel's callbacks API. Something to think about for the future.

Absolutely! I added a commit with a dedicated callback. Will make sense to handle the callback thing in a different PR.